client/src/controllers/BulkController.ts

```diff
@@ -0,0 +1,69 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Controller that adds the ability to toggle some or all the elements, ignoring the disabled ones
+ * Additionaly, it checks if all non-disabled items are selected or not. If they are, then select all
+ *
+ * @example
+ * <div data-controller="w-bulk">
+ *   <input type="checkbox" data-w-bulk-target="all">
+ *
+ *   <div>
+ *     <input type="checkbox" data-w-bulk-target="item" disabled>
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *   </div>
+ *
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="false">Clear all</button>
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="true">Select all</button>
+ * </div>
+ */
+export class BulkController extends Controller {
```

Comment:
```suggestion
/**
 * Adds the ability to collectively toggle a set of (non-disabled) checkboxes.
 *
 * @example
 * <div data-controller="w-bulk">
 *   <input type="checkbox" data-action="w-bulk#toggleAll" data-w-bulk-target="all">
 *   <div>
 *     <input type="checkbox" data-action="w-bulk#change" data-w-bulk-target="item" disabled>
 *     <input type="checkbox" data-action="w-bulk#change" data-w-bulk-target="item">
 *     <input type="checkbox" data-action="w-bulk#change" data-w-bulk-target="item">
 *   </div>
 *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="false">Clear all</button>
 *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="true">Select all</button>
 * </div>
 */
export class BulkController extends Controller<HTMLElement> {
```

* Let's clean up this example code a bit and ensure it's correct (remember we need the data-actions)
* Simplify the JSDoc description
* Add the generic type of this controller's controlled element, it's not used for the code but just nice to have